### PR TITLE
init

### PR DESCRIPTION
**Pull request by connor4312**
See the readme for deets

### CODE REVIEW

File: incrpx.lua [hunk 1]
```diff
@@ -0,0 +1,14 @@
+--
```
Comment (SimonSchick) [inline incrpx.lua]: FYI: ```--[[ ]]``` for multi line comments.

File: index.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    return value;
+}
+
+/**
+ * Limiter is the rate limiter parent client. You can define rules on it.
+ */
+export class Limiter {
+
+    private rules: { [name: string]: Rule<any> } = Object.create(null);
```
Comment (SimonSchick) [inline index.ts]: Might as well use `Map`

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "noImplicitAny": true,
+        "noImplicitReturns": true,
+        "noImplicitThis": true,
+        "noUnusedLocals": true,
+        "noUnusedParameters": true,
+        "strictNullChecks": true,
+        "target": "es5",
```
Comment (SimonSchick) [inline tsconfig.json]: We can probably es6 here, unless you want this to work in browsers.

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (SimonSchick) [inline tsconfig.json]: minor: redundant nl

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Missing `.npmignore` and ~~license~~ nvm already present

File: tsconfig.json (hunk 3 above)
Comment (connor4312) [reply tsconfig.json]: Node 4 is still LTS 😛

File: index.ts (hunk 2 above)
Comment (connor4312) [reply index.ts]: Plain objects are about 3x faster than string-keyed maps. Rate limiters tend to be very hot paths, so I would tend to favor an object here.

Comment (SimonSchick) [reply index.ts]: Interesting, I remember reading v8 Maps were pretty fast, too bad.

Comment (connor4312) [reply index.ts]: They are quite fast! Around 50 million ops/sec. Plain objects, though, are more like 150 million ops/sec

**Review by JamyDev — APPROVED** [review]

**Comment by SimonSchick** [comment]
Actually, I'd suggest moving all source and tests into a `src` folder, we use this pattern in other TS packages and it makes things pretty smooth as the build folder is kept seperately in `dist`.

**Comment by connor4312** [comment]
Swapped things around @SimonSchick . A bit cleaner now 😄

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+language: node_js
+node_js:
```
Comment (SimonSchick) [inline .travis.yml:2]: Minor: You wanted to support v4, might as well test for it.

**Review by SimonSchick — APPROVED** [review]

**Pull request merged**